### PR TITLE
feat(nuxt): pass server logs to client

### LINKED ISSUE #13470: Show SSR logs in client console (including errors and warns)
- via consola v3 - WIP: https://github.com/unjs/consola/pull/135

### LINKED ISSUE #25260: DX: Forward SSR logs in development back to the browser console
### Describe the feature

Similar to Nuxt 2 feature: https://nuxt.com/blog/nuxtjs-from-terminal-to-browser

58566291-a3396700-8230-11e9-9dd6-09c3ff8578d2

This is a major DX improvement IMO as many developers don't necessary know that they have to look at the terminal when refreshing the page.

And since `console.log` is a popular way to debug JS application...

### Nuxt 2 code:

Initial PR back then in 2019: https://github.com/nuxt/nuxt/pull/5673

During SSR: https://github.com/nuxt/nuxt/blob/246a9227267d6933f18b7cbdd1e3790edbdff602/packages/vue-renderer/src/renderers/ssr.js#L77-L109

On client-side: https://github.com/nuxt/nuxt/blob/246a9227267d6933f18b7cbdd1e3790edbdff602/packages/vue-app/template/client.js#L58-L67

### Additional information

- [ ] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

### PR DESCRIPTION
**Pull request by danielroe**
### 🔗 Linked issue

resolves https://github.com/nuxt/nuxt/issues/25260
resolves https://github.com/nuxt/nuxt/issues/13470

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This PR adds a new feature (can be disabled by setting `features.devLogs` to `false`) which records all console events on the server (nitro runtime + vue runtime). It bundles up logs to the next request, and then continues to stream them to the client. Any logs that don't happen on the client are displayed in the console.

A hook (which I envision being consumed by devtools - cc: @antfu) is called whenever new logs are available.

You can currently set `features.devLogs` to `silent` to disable them being output to the browser console while still being able to consume them in devtools, or elsewhere. I'm wondering if it makes sense for this to be the default, so things are less 'noisy'... (Thoughts very much welcome!)

Here's a demo of logs being consumed by a custom hook + [`vue-sonner`](https://vue-sonner.vercel.app/):

https://github.com/nuxt/nuxt/assets/28706372/95b72eee-71eb-4bf3-bcee-d0a7c4557b60

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have added tests (if possible).
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/nuxt/types.d.mts [hunk 1]
```diff
@@ -27,6 +28,7 @@ declare module 'nitropack' {
     experimentalNoScripts?: boolean
   }
   interface NitroRuntimeHooks {
+    'dev:ssr-logs': (ctx: { logs: LogObject[], path: string }) => void | Promise<void>
```
Comment (pi0) [inline packages/nuxt/types.d.mts:31]: (when adding feature to nitro, i would probably use `dev:logs` so should be fine only mentioning the idea)

Comment (danielroe) [reply packages/nuxt/types.d.mts:31]: would you like me to make a pr to add this to nitro?

Comment (pi0) [reply packages/nuxt/types.d.mts:31]: surely would love if you like to help. i probably make a tracker issue once enabled ALS first globally which i desperately hope to do 😆

**Review by pi0 — APPROVED** [review]

**Review by atinux — APPROVED** [review]

**Pull request merged**